waspc/src/Wasp/Analyzer/Evaluator/EvaluationError.hs

```diff
@@ -62,57 +59,70 @@ data EvaluationParseError
 mkEvaluationError :: Ctx -> EvaluationError' -> EvaluationError
 mkEvaluationError ctx e = EvaluationError $ WithCtx ctx e
 
+concatCtxMessages :: [String] -> String
+concatCtxMessages [] = ""
+concatCtxMessages msgChain = prefix ++ foldr1 appendMsg msgChain
+  where
+    prefix = "-> "
+    appendMsg currMsg = (++) (currMsg ++ ":\n") . indent 2 . (prefix ++)
```

Comment:
Ok, so this is part that is common between TypeError and EvaluationError, and it knows notihng about them, it just handles strings which are context errors.

I think this one we can externalize right now, without much thinking. I am not sure where to put it hm -> Analyzer/ErrorCtx.hs ?

Reply:
Yes, but the function below (`getErrorMessageAndCtx`) is pretty similar as well. I could extract that one too, although it would require a bit of rearranging.

Reply:
Yeah but it has different types in the signature right? And it is so short that I wouldn't bother with it for now. Unless you have some interesting idea, but I think you won't gain much by trying to extract it.